app.go

```diff
@@ -224,3 +226,75 @@ func executeQuote(cmd command) bool {
 	// send the quote to the user
 	return true
 }
+
+func executeBuy(cmd command) bool {
+	//Gotta check users money and add a reserved portion
+	account := accountStore.GetAccount(cmd.UserID)
+
+	if account == nil {
+		log.Noticef("User %s does not have an account", account)
+	}
+
+	stockSymbol := cmd.Args[0]
+	dollarAmount, err := currency.NewFromString(cmd.Args[1])
+	//dollarAmount, err := strconv.ParseFloat(cmd.Args[1], 64)
+	if err != nil {
+		log.Noticef("Dollar amount %s is invalid", cmd.Args[1])
+		return false
+	}
+	//User wants to buy y worth of x shares.
+	userQuote, err := quotecache.GetQuote(cmd.UserID, stockSymbol)
+
+	if err != nil {
+		log.Noticef("Quote of stock %s for user %s is invalid", stockSymbol, cmd.UserID)
+		return false
+	}
+
+	wholeShares, cashRemainder := userQuote.Price.FitsInto(dollarAmount)
+
+	if wholeShares == 0 {
+		log.Notice("Amount specified to buy less than single stock unit")
```

Comment:
IMO `log.Debug` should be used liberally to capture the state of the command being executed.